libnmstate/nm/dns.py

```diff
@@ -118,3 +129,41 @@ def get_dns_config_iface_names(acs_and_ipv4_profiles, acs_and_ipv6_profiles):
         if ip_profile.props.dns or ip_profile.props.dns_search:
             iface_names.append(nm_ac.ActiveConnection(ac).devname)
     return iface_names
+
+
+def choose_interfaces_for_name_servers(iface_routes, servers):
+    """
+    Find interfaces to store the DNS configurations:
+        * Interface with static gateway configured.
+    Return two tuples like below
+        (iface_name, ipv4_servers, dns_priority),
+        (iface_name, ipv6_servers, dns_priority)
+    The iface_name will be None if failed to find proper interface.
+    """
+    ipv6_servers = [server for server in servers
+                    if iplib.is_ipv6_address(server)]
+    ipv4_servers = [server for server in servers
+                    if not iplib.is_ipv6_address(server)]
+    ipv4_iface = _get_static_gateway_iface(Interface.IPV4, iface_routes)
+    ipv6_iface = _get_static_gateway_iface(Interface.IPV6, iface_routes)
+    ipv4_priority = DNS_DEFAULT_PRIORITY_FOR_STATIC
+    ipv6_priority = DNS_DEFAULT_PRIORITY_FOR_STATIC
+    if ipv4_servers and ipv6_servers and \
+       servers == [ipv4_servers[0], ipv6_servers[0]]:
+        # Since IPv6 take priority over IPv4, we need use DNS priority
```

Comment:
we need *to* use ... when *the* desire config...
I do not understand `pure IPv4`. I guess you mean that you must enforce the priority only when IPv4 is placed before IPv6.

But why would you even care? You can just set the priority based on who came first in the order, no?
Won't that solve it and avoid this special checking?

Reply:
Yes. Have simplify thing. I will use that way.

Reply:
That does not simplify code, it need to all the cases of servers which need more code that 
current layout. The current layout just need to take care the special case -- `ipv4 before ipv6`.

Reply:
I thought the priority is per the order the entries are in the list,
therefore my assumption was that this is a unified solution:
```
servers_priority = [server_priority for server_priority in enumerate(servers, DNS_ORDERING_BASE_PRIORITY)]
```

i.e. the priority is set per the order of the given servers.